sensor/tests/complianceoperator/crd_test.go

```diff
@@ -0,0 +1,52 @@
+package complianceoperator
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/sensor/tests/helper"
+	"github.com/stretchr/testify/require"
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+)
+
+func Test_ComplianceOperatorCRDsDetection(t *testing.T) {
+	t.Setenv(env.ConnectionRetryInitialInterval.EnvVar(), "1s")
+	t.Setenv(env.ConnectionRetryMaxInterval.EnvVar(), "2s")
+
+	c, err := helper.NewContextWithConfig(t, helper.Config{
+		InitialSystemPolicies: nil,
+		CertFilePath:          "../../../tools/local-sensor/certs/",
+	})
+	t.Cleanup(c.Stop)
+
+	require.NoError(t, err)
+
+	c.RunTest(t, helper.WithTestCase(func(t *testing.T, testContext *helper.TestContext, _ map[string]k8s.Object) {
+		// Wait for the first sync
+		testContext.WaitForSyncEvent(t, 10*time.Second)
+		testContext.GetFakeCentral().ClearReceivedBuffer()
+
+		// Create the Compliance Operator CRDs
+		deleteCRDsFn, err := testContext.ApplyWithManifestDir(context.Background(), "../../../tests/complianceoperator/crds", "*")
+		require.NoError(t, err)
+
+		// Just in case the tests fails we don't want to leave resources behind
+		defer func() {
+			utils.IgnoreError(deleteCRDsFn)
+		}()
+
+		// Sensor should sync again after the CRDs are detected
+		testContext.WaitForSyncEvent(t, 10*time.Second)
```

Comment:
[nit] These `WaitForSyncEvent` functions were previously used as a way to force stop the test if a precondition is not met (i.e. connection setup correctly). 

In your case, these are actual assertions (the connection was indeed restarted). But I think the failed test result will look more like fatal error in the test harness. 

Could we maybe offer an extra parameter to this function (or perhaps a new alternative `WaitForSyncEventf`) that allows formatting strings to be passed? This will allow you to add an assertion message which will make the failure more clear. 

The ideal code (in my opinion) should look something like:

```suggestion
		testContext.WaitForSyncEvent(t, 10*time.Second, "expected restart connection after CRDs are removed")
```

Reply:
Makes sense. I added a new function `WaitForSyncEventf` like you suggested with an extra parameter for the error message.